crates/holochain_cascade/src/lib.rs

```diff
@@ -774,49 +748,65 @@ impl CascadeImpl {
         // Is this bad because we will not go back to the network until our
         // cache is cleared. Could someone create an attack based on this fact?
 
-        if let GetStrategy::Local = options.strategy {
-            // Only return what is in the database.
-            return self.cascading(query.clone()).await;
-        }
-
-        // If we are not in the process of authoring this hash or its
-        // authority we need a network call.
-        let authoring = self.am_i_authoring(&action_hash.clone().into())?;
-        let authority = self.am_i_an_authority(action_hash.clone().into()).await?;
-        if !(authoring || authority) {
-            self.fetch_record(action_hash.into(), options.into())
-                .await?;
-        }
-
-        // Check if we have the data now after the network call.
-        self.cascading(query).await
+        self.get_with_query(query, action_hash.into(), options)
+            .await
     }
 
-    #[cfg_attr(feature = "instrument", tracing::instrument(skip(self, options)))]
     /// Returns the oldest live [Record] for this [EntryHash] by getting the
     /// latest available metadata from authorities combined with this agents authored data.
+    #[cfg_attr(feature = "instrument", tracing::instrument(skip(self, options)))]
     pub async fn dht_get_entry(
         &self,
         entry_hash: EntryHash,
         options: GetOptions,
     ) -> CascadeResult<Option<Record>> {
         let query: GetLiveEntryQuery = self.construct_query_with_data_access(entry_hash.clone());
 
-        if let GetStrategy::Local = options.strategy {
-            // Only return what is in the database.
-            return self.cascading(query.clone()).await;
-        }
+        self.get_with_query(query, entry_hash.into(), options).await
+    }
 
-        // If we are not in the process of authoring this hash or its
-        // authority we need a network call.
-        let authoring = self.am_i_authoring(&entry_hash.clone().into())?;
-        let authority = self.am_i_an_authority(entry_hash.clone().into()).await?;
-        if !(authoring || authority) {
-            self.fetch_record(entry_hash.into(), options.into()).await?;
+    async fn get_with_query<Q, O>(
```

Comment:
Feel free to change the name, but we have a lot of trouble at-a-glance knowing if something is going to short-circuit locally or not, so might be nice to call it out explicitly in the function name to make it easy to refactor if we need to in the future.

```suggestion
    async fn get_local_first_with_query<Q, O>(
```

Reply:
Or maybe I'm not understanding that `self.cascading` call below. This still depends on the GetOptions, doesn't it?

Reply:
Agreed and done! https://github.com/holochain/holochain/pull/4800/commits/b2a375efd0929d32e0495379b3258bc255c15394